### PR TITLE
RSpec fingerprint

### PR DESCRIPTION
**Pull request by CamAnNguyen**
- Refactor fingerprint model
- Add rspec for fingerprint
- Edit an old migrate to stop populate a deleted column (molecule.num_set_bits), so that it does not take much time if migrate from the beginning

### CODE REVIEW

File: app/models/fingerprint.rb [hunk 1]
```diff
@@ -2,7 +2,31 @@ class Fingerprint < ActiveRecord::Base
...
+            numericality: {
+              only_integer: true,
+              greater_than: 0,
+              less_than: 64,
+              message: "must be integer between 0 and 64"
+            }
+
+  after_create :check_num_set_bits
+
+  # before_save :find_or_create_by_molfile
```
Comment (Sergeykot) [inline app/models/fingerprint.rb]: odd commented codce

**Comment by Sergeykot** [comment]
only one thing

https://github.com/ComPlat/chemotion_ELN/pull/122/commits/7a02e1bb44167ddbd38c57eeb493e05e2a56b9a0#r75272302

**Comment by CamAnNguyen** [comment]
Comment removed.

**Comment by Sergeykot** [comment]
:+1:

**Pull request merged**